src/execution_plan/optimizations/traverse_order.c

```diff
@@ -296,12 +295,12 @@ static void _select_entry_point(AlgebraicExpression *ae, rax *filtered_entities,
  * we pick the order in which the expressions will be evaluated
  * taking into account filters and transposes.
  * exps will reordered. */
-void orderExpressions(AlgebraicExpression **exps, uint exp_count, const FT_FilterNode *filters,
-					  rax *bound_vars) {
+void orderExpressions(QueryGraph *qg, AlgebraicExpression **exps, uint exp_count,
```

Comment:
I think the first argument should be `exps` `qg` should come before `filters` or after `bound_vars`
Moreover I think that a function which had its signature modified to accommodate QueryGraph if that function doesn't uses the QueryGraph directly but passes it over to some other function should get the QueryGraph from TLS, this is just to keep the signature shorter and makes for a better reasoning when looking at the function's sig (in my opinion).

Reply:
Do you _really_ want to store the QueryGraph in TLS, though, given the complications we've already seen with the AST in TLS? All of the AST resetting logic would have to be replicated for the QueryGraph.